flink-tests/src/test/java/org/apache/flink/test/checkpointing/RescalingITCase.java

```diff
@@ -876,6 +887,8 @@ public void run(SourceContext<Integer> ctx) throws Exception {
                     break;
                 }
             }
+
+            canFinishLatch.await();
```

Comment:
And the second comment was lost...

We could move the await inside the loop, outside the sync block. Now it's rather a zombie then a living source.

Reply:
Not really... We won't have any progress if we move it into the loop. What is the problem with a "zombie" source? The only difference is in the internal `running` flag, which is not visible outside the source anyway.

Reply:
Well if we are not producing any data anyways anymore, then there is no difference.